bentoml/_internal/configuration/containers.py

```diff
@@ -434,5 +442,32 @@ def tracing_excluded_urls(
     remote_runner_mapping = providers.Static[t.Dict[str, str]]({})
     plasma_db = providers.Static[t.Optional["ext.PlasmaClient"]](None)
 
+    @providers.SingletonFactory
+    @staticmethod
+    def duration_buckets(
+        metrics: t.Dict[str, t.Any] = Provide[config.api_server.metrics],
+    ) -> t.Tuple[float, ...]:
+        """
+        Returns a tuple of duration buckets in seconds. If not explicitly configured,
+        the Prometheus default is returned; otherwise, a set of exponential buckets
+        generated based on the configuration is returned.
+        """
+        from ..utils.metrics import DEFAULT_BUCKET
+        from ..utils.metrics import exponential_buckets
+
+        if "duration" not in metrics:
+            return DEFAULT_BUCKET
+        else:
+            duration: t.Dict[str, float] = metrics["duration"]
```

Comment:
```suggestion
            duration: dict[str, float] = metrics["duration"]
```
if 'duration' keys does not exist, this will throw `KeyError`

Reply:
Will the previous if statement check for it?

Reply:
you are correct.